File: src/main/java/dev/vxrp/bot/commands/CommandListener.java
package dev.vxrp.bot.commands;

import dev.vxrp.bot.commands.help.HelpCommand;
import dev.vxrp.bot.commands.templates.TemplateCommand;
import dev.vxrp.util.Enums.DCColor_DEPRECATED;
import dev.vxrp.util.colors.ColorTool;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Objects;

public class CommandListener extends ListenerAdapter {
    private final Logger logger = LoggerFactory.getLogger(CommandListener.class);

    @Override
    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
        String template = Objects.requireNonNull(event.getOption("template")).getAsString();

        // Help
        if (event.getFullCommandName().equals("help")) {
            HelpCommand.pasteHelp(event);
        }

        // Template
        if (event.getFullCommandName().equals("template") && template.equals("rules")) {
            TemplateCommand.templateRules(event);
        }
        if (template.equals("support")) {
            TemplateCommand.templateSupport(event);
        }

        if (template.equals("notice_of_departure")) {
            TemplateCommand.templateNoticeOfDeparture(event);
        }
        if  (template.equals("regulars")) {
            TemplateCommand.templateRegulars(event);
        }
        logger.info("User {} executed command {} with args '{}'", ColorTool.apply(DCColor_DEPRECATED.GREEN, event.getUser().getGlobalName()), event.getFullCommandName() , template);
    }
}

File: src/main/kotlin/dev/vxrp/bot/commands/CommandListener.kt
package dev.vxrp.bot.commands

import dev.minn.jda.ktx.events.listener
import dev.vxrp.bot.commands.data.CustomCommand
import dev.vxrp.bot.commands.help.HelpCommand
import dev.vxrp.configuration.loaders.Config
import dev.vxrp.configuration.loaders.Translation
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class CommandListener(val api: JDA, val config: Config, val translation: Translation) : ListenerAdapter() {
    init {
        api.listener<SlashCommandInteractionEvent> { event ->
            val commandManager = CommandManager(config, "/configs/commands.json")

            val commandList = mutableListOf<CustomCommand>()

            commandList.addAll(commandManager.query().commands)
            commandList.addAll(commandManager.query().statusCommands)

            for (command in commandList) {
                if (command.name != event.fullCommandName) continue
                checkInheritance(command.inherit, event)
                break
            }
        }
    }

    private fun checkInheritance(inherit: String, event: SlashCommandInteractionEvent) {
        when (inherit) {
            "commands.help.default" -> helpCommand(event)

            "commands.template.default" -> templateCommand()

            "commands.notice_of_departure.default" -> noticeOfDepartureCommand()

            "commands.regulars.default" -> regularsCommand()

            "status_commands.status.default" -> statusCommand()
        }
    }

    private fun helpCommand(event: SlashCommandInteractionEvent) {
        HelpCommand(translation).pasteHelpMenu(event)
    }

    private fun templateCommand() {

    }

    private fun noticeOfDepartureCommand() {

    }

    private fun regularsCommand() {

    }

    private fun statusCommand() {

    }
}